collector/benthos/output/openmeter.go

```diff
@@ -176,3 +227,71 @@ func (o *openmeterOutput) WriteBatch(ctx context.Context, batch service.MessageB
 func (o *openmeterOutput) Close(_ context.Context) error {
 	return nil
 }
+
+func (o *openmeterOutput) UpdateMessageSpan(ctx context.Context, msg *service.Message) {
+	msgSpan := trace.SpanFromContext(msg.Context())
+	if msgSpan == nil {
+		o.logger.Debug("no span found for message")
+
+		return
+	}
+	msgSpan.AddLink(trace.LinkFromContext(ctx))
+
+	var spanAttrs []attribute.KeyValue
+	if o.tracingAttrsMap != nil {
+		spanAttrsMsg, err := msg.BloblangQuery(o.tracingAttrsMap)
+		if err != nil {
+			o.logger.Debugf("failed to extract tracing attributes from message: %v", err)
+		}
+
+		var spanAttrsVal any
+		if spanAttrsMsg != nil {
+			if spanAttrsVal, err = spanAttrsMsg.AsStructured(); err != nil {
+				o.logger.Debugf("failed to construct structured tracing data from message: %v", err)
+			}
+		}
+
+		if spanAttrsVal != nil {
+			spanAttrMap, ok := spanAttrsVal.(map[string]interface{})
+			if !ok {
+				o.logger.Debugf("tracing attributes mapping resulted in a non-object mapping: %T", spanAttrsVal)
+			}
+
+			for k, v := range spanAttrMap {
+				attrs := toAttrs(k, v)
+				spanAttrs = append(spanAttrs, attrs...)
+			}
+		}
+	}
```

Comment:
Maybe return early instead of embedding ifs into each other?

Reply:
Fixed.